source/common/upstream/upstream_impl.cc

```diff
@@ -727,7 +728,8 @@ ClusterInfoImpl::ClusterInfoImpl(
       peekahead_ratio_(PROTOBUF_GET_WRAPPED_OR_DEFAULT(config.preconnect_policy(),
                                                        predictive_preconnect_ratio, 0)),
       per_connection_buffer_limit_bytes_(
-          PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, per_connection_buffer_limit_bytes, 1024 * 1024)),
+          PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, per_connection_buffer_limit_bytes,
+                                          DefaultsProfileSingleton::get().cluster.max_buffer_size)),
```

Comment:
It is unclear to me how different profile can be used in this case.
Maybe I'm wrong but shouldn't this be a call to `CurrentProfileSingleton` that returns whatever selected profile (where `DefaultProfileSingleton` is one of them)?